File: webapi/Models/Subscription.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System;
using System.Text.Json.Serialization;

namespace webapi.Models {
    public class Subscription {
        public Guid Id { get; set; }
        [JsonIgnore]
        public Guid UserId { get; set; }
        public string Name { get; set; }
        public decimal Amount { get; set; }
        public DateTime RenewalDate { get; set; }
        public Subscription() {

        }
        public Subscription(Data.Models.Subscription pSubscription) {
            this.Id = pSubscription.Id;
            this.RenewalDate = pSubscription.RenewalDate;
            this.Name = pSubscription.Name;
            this.Amount = pSubscription.Amount;
            this.UserId = new Guid(pSubscription.UserId);
        }
    }
}

File: clientapp/src/domains/models/Subscription.ts
export interface ISubscription {
  id: string;
  name: string;
  amount: number;
  renewalDate: Date;
  userId?: string;
}

export class Subscription {
  public readonly id: string;
  public readonly name: string;
  public readonly amount: number;
  public readonly renewalDate: Date;

  constructor(data: ISubscription) {
    this.id = data.id;
    this.name = data.name;
    this.amount = data.amount;
    this.renewalDate = data.renewalDate;
  }
}
